src/functions/docs.ts

```diff
@@ -28,33 +30,65 @@ function resolveElementString(element: DocElement, doc: Doc): string {
 	return `${parts.join('')}\n${description}`;
 }
 
-function resolveResultString(results: DocElement[]): string {
-	const res = [
-		`${PREFIX_FAIL} No match. Here are some search results:`,
-		...results.map((res) => `• **${escapeMDLinks(res.link ?? '')}**`),
-	];
-	return res.join('\n');
+function buildSelectOption(result: DocElement, emojiId: string) {
+	return {
+		label: result.formattedName,
+		value: result.formattedName,
+		description: truncate(result.formattedDescription ?? result.description ?? 'No description found', 47),
+		emoji: {
+			id: emojiId,
+		},
+	};
 }
 
-export async function djsDocs(res: Response, source: string, query: string, target?: string): Promise<Response> {
-	query = query.trim();
-	const doc = await Doc.fetch(source, { force: true });
+export function fetchDocResult(source: string, doc: Doc, query: string, user?: string, target?: string): string | null {
 	const element = doc.get(...query.split(/\.|#/));
-	const icon = source === 'master' ? EMOJI_DJS_DEV : EMOJI_DJS;
+	if (!element) return null;
+	const icon = formatEmoji(source === 'master' ? EMOJI_ID_DJS_DEV : EMOJI_ID_DJS);
+	const messageParts = [];
+	if (user || target) {
```

Comment:
```suggestion
	if (user ?? target) {
```
could technically do this